src/main/java/com/work/management/service/employee/impl/EmployeeServiceImpl.java

```diff
@@ -51,4 +51,19 @@ public EmployeeDto getEmployeeByUserName(String username) {
     BeanUtils.copyProperties(employee.get(), employeeDto);
     return employeeDto;
   }
+
+  @Override
+  public EmployeeDto updateEmployeeEntity(EmployeeDto employeeDto) {
+    Optional<Employee> employee = employeeRepository.findById(employeeDto.getId());
+    if (!employee.isPresent()) {
+      ExceptionUtils
+          .throwEntityNotFoundException(
+              String
+                  .format("Employee with Id %d doesn't exists.", employeeDto.getId()));
+    }
+
+    BeanUtils.copyProperties(employeeDto, employee.get());
+    employeeRepository.save(employee.get());
```

Comment:
I don't think this work as intended...there might be still older values which are not getting updated in employeeDto be in Db...just check once...
You might need to create a new Entity object and then copy employeeDto fields to the entity.